ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java

```diff
@@ -376,8 +378,8 @@ protected void executeProject(JBallerinaBackend jBallerinaBackend) throws Invoke
             // First run configure initialization
             // TODO: (#28662) After configurables can be supported, change this to that file location
             invokeMethodDirectly(classLoader, CONFIGURE_INIT_CLASS_NAME, CONFIGURE_INIT_METHOD_NAME,
-                    new Class[]{String[].class, Path[].class, String.class}, new Object[]{new String[]{},
-                            new Path[]{}, null});
+                    new Class[]{Map.class, String[].class, Path[].class, String.class},
+                    new Object[]{new HashMap<>(), new String[]{}, new Path[]{}, null});
```

Comment:
What are the changes done for config init?

Reply:
Following changes were applied to the `$configureInit` design,
- We call `LaunchUtils.addModuleConfigData(var0, _init.$moduleName, var4)` only for the current module.
- For first level dependency modules(immediate imports) of the current module we call their respective `$configureInit` methods. Which will call their respective `LaunchUtils.addModuleConfigData(var0, _init.$moduleName, var4)` methods.
- We added an `if check` to stop duplicate invocations of `$configureInit` when there are common dependencies between modules.

All changes in this PR can be found [in this document](https://docs.google.com/document/d/1yjTv5e38wfYhYoJgmZgb2HH7KXYvOt06c4-GDBsP0wU/edit?usp=sharing).